x-pack/plugins/uptime/public/lib/helper/component_test_helpers.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactElement } from 'react';
+import { render as reactRender } from '@testing-library/react';
+import { Router } from 'react-router-dom';
+import { MemoryHistory } from 'history/createMemoryHistory';
+import { createMemoryHistory, History } from 'history';
+import { I18nProvider } from '@kbn/i18n/react';
+import { mountWithIntl, renderWithIntl, shallowWithIntl } from '@kbn/test/jest';
+import { coreMock } from 'src/core/public/mocks';
+import {
+  KibanaContextProvider,
+  KibanaServices,
+} from '../../../../../../src/plugins/kibana_react/public';
+import { MountWithReduxProvider } from './helper_with_redux';
+import { AppState } from '../../state';
+
+interface KibanaProps {
+  services?: KibanaServices;
+}
+
+interface KibanaProviderOptions {
+  coreOptions?: any;
```

Comment:
Are we able to reference `CoreStart` here? As I understand it we use this field to let people supply their own `coreOptions`, which we spread overtop of the default mock we create. So could we type this as `Partial<CoreStart>`, so the caller could supply just one or two fields if needed, or a full mock if they wanted?

```suggestion
  coreOptions?: Partial<CoreStart>;
```